### PR TITLE
[Android] Enable chip_enable_schema_check for android platform.

### PR DESCRIPTION
**Pull request by chulspro**
Signed-off-by: sanghyukko <sanghyuk.ko@samsung.com>
Signed-off-by: Charles Kim <chulspro.kim@samsung.com>

#### Problem
MAC/Linux already enabled detail info for certification,
We also want to enable detail info of cluster message on Android platform.

#### Change overview
Enable android platform os on GN build

#### Testing
Example message detail info after changes.
[1646313566.134637][8200:8200] CHIP:DMG: IM WH moving to [Initialized]
[1646313566.134745][8200:8200] CHIP:DMG: WriteRequestMessage =
[1646313566.134804][8200:8200] CHIP:DMG: {
[1646313566.134861][8200:8200] CHIP:DMG:        timedRequest = false,
[1646313566.134920][8200:8200] CHIP:DMG:        AttributeDataIBs =
[1646313566.134998][8200:8200] CHIP:DMG:        [
[1646313566.135058][8200:8200] CHIP:DMG:                AttributeDataIB =
[1646313566.135128][8200:8200] CHIP:DMG:                {
[1646313566.135195][8200:8200] CHIP:DMG:                        AttributePathIB =
[1646313566.135281][8200:8200] CHIP:DMG:                        {
[1646313566.135349][8200:8200] CHIP:DMG:                                Endpoint = 0x0,
[1646313566.135429][8200:8200] CHIP:DMG:                                Cluster = 0x28,
[1646313566.135519][8200:8200] CHIP:DMG:                                Attribute = 0x0000_0005,
[1646313566.135599][8200:8200] CHIP:DMG:                        }
[1646313566.135682][8200:8200] CHIP:DMG:
[1646313566.135764][8200:8200] CHIP:DMG:                                Data = "te5new",
[1646313566.135877][8200:8200] CHIP:DMG:                },
[1646313566.135955][8200:8200] CHIP:DMG:
[1646313566.136013][8200:8200] CHIP:DMG:        ],
[1646313566.136083][8200:8200] CHIP:DMG:
[1646313566.136141][8200:8200] CHIP:DMG:        moreChunkedMessages = false,
[1646313566.136205][8200:8200] CHIP:DMG:        InteractionModelRevision = 1
[1646313566.136260][8200:8200] CHIP:DMG: },

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

File: BUILD.gn [hunk 1]
```diff
@@ -35,7 +35,7 @@ import("//src/crypto/crypto.gni")
 if (current_toolchain != "${dir_pw_toolchain}/default:default") {
   declare_args() {
     chip_enable_python_modules =
-        (current_os == "mac" || current_os == "linux") &&
+        (current_os == "mac" || current_os == "linux" || current_os == "android") &&
         (host_cpu == "x64" || host_cpu == "arm64" || host_cpu == "arm")
     enable_pylib = false
```
Comment (msandstedt) [inline BUILD.gn:40]: I don't know the gn build system well enough to evaluate whether this is correct.  Is it true that setting `enable_pylib` false enables `chip_enable_schema_check`?

Also, is there a way to override this parameter for the build?  If so, would it be appropriate to do so?  I am only asking the question.  If this is about uniformity and providing a sensible default, I'm sure it makes sense.

Comment (andy31415) [reply BUILD.gn:40]: The logic seems to say "enable python modules if we compile for android" ... is this really what we want?

**Review by andy31415 — DISMISSED** [review]
Change is non-obvious: it says enabling chip_enable_schema_check but instead it changes the default for chip_enable_python_modules

**Comment by chulspro** [comment]
> Change is non-obvious: it says enabling chip_enable_schema_check but instead it changes the default for chip_enable_python_modules

I think we change different file during the commit migration from our code to contribution code because of conflicts. I will fix and update again.

File: BUILD.gn (hunk 1 above)
Comment (chulspro) [reply BUILD.gn:40]: I'm sorry It is different file changes during the patch migration. we want to change src/app/BUILD.gn not root.

Comment (chulspro) [reply BUILD.gn]: We have changed commit again. Please review this patch again @andy31415

**Review by bzbarsky-apple — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Comment by chulspro** [comment]
> Change is non-obvious: it says enabling chip_enable_schema_check but instead it changes the default for chip_enable_python_modules

Can you check again our final commit? @andy31415

**Review by msandstedt — APPROVED** [review]

**Pull request merged**